test/suites/edgexfoundry/proxy_test.go

```diff
@@ -123,3 +134,42 @@ func generateCerts(tmpDir string) (string, string) {
 
 	return caCertFile, caKeyFile
 }
+
+// waitCertInstall checks if certificate is fully installed
+// up to a maximum retry number
+func waitCertInstall(t *testing.T, caKey string, maxRetry int) {
+
+	type ResponseBody struct {
+		Data []struct {
+			Key string
+		}
+	}
+
+	res := &ResponseBody{}
+
+	for i := 1; ; i++ {
+		t.Logf("Checking certificate installation. Retry %d/%d", i, maxRetry)
+
+		client := &http.Client{}
+		req, err := http.NewRequest("GET", "http://localhost:8001/certificates", nil)
+		require.NoError(t, err)
+		resp, err := client.Do(req)
+		require.NoError(t, err)
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		err = json.Unmarshal(data, res)
```

Comment:
ReadAll's error isn't handled.
You can combine ReadAll and json unmarshal: `json.NewDecoder(resp.Body).Decode(res)`